### PR TITLE
Bug 1926128: Dvm proxy support

### PR DESCRIPTION
**Pull request by pranavgaikwad**
This PR adds support for using TCP_PROXY in Stunnel enabling Rsync work behind a TCP proxy.

It also adds new variables to allow configuring CA verification in Stunnel. 

Please also review: https://github.com/konveyor/mig-operator/pull/599

### CODE REVIEW

**Comment by shawn-hurley** [comment]
Sorry if I missing this or if it is in a different PR but does this also split the stunnel into each rsync pod?

**Comment by pranavgaikwad** [comment]
@shawn-hurley This is only the proxy stuff that needs to go in 1.4.2. The other thing I think is de-scoped to 1.4.3

File: pkg/controller/directvolumemigration/stunnel.go [hunk 1]
```diff
@@ -81,6 +107,27 @@ data:
...
 `
 
+// generateStunnelProxyConfig loads stunnel proxy configuration from app settings
+func (t *Task) generateStunnelProxyConfig() (stunnelProxyConfig, error) {
+	var proxyConfig stunnelProxyConfig
+	tcpProxyString := settings.Settings.DvmOpts.StunnelTCPProxy
+	if tcpProxyString != "" {
+		url, err := url.Parse(tcpProxyString)
+		if err != nil {
+			t.Log.Error(err, "failed to parse TCP_PROXY setting")
```
Comment (djwhatle) [inline pkg/controller/directvolumemigration/stunnel.go:117]: Any reason why we aren't using liberr.Wrap here?

File: pkg/controller/directvolumemigration/stunnel.go [hunk 2]
```diff
@@ -81,6 +107,27 @@ data:
...
+			return proxyConfig, err
```
Comment (djwhatle) [inline pkg/controller/directvolumemigration/stunnel.go:118]: Any reason why we aren't using liberr.Wrap here?

**Review by djwhatle — APPROVED** [review]
I'm not the person to review this for proxy correctness (not familiar with the problem area), but visually PR looks good except for that one liberr.wrap omission.

File: pkg/controller/directvolumemigration/stunnel.go (hunk 2 above)
Comment (pranavgaikwad) [reply pkg/controller/directvolumemigration/stunnel.go]: Missed it by mistake, fixed now

File: pkg/controller/directvolumemigration/stunnel.go [hunk 3]
```diff
@@ -70,8 +97,8 @@ data:
     pid =
     socket = l:TCP_NODELAY=1
     socket = r:TCP_NODELAY=1
-    sslVersion = TLSv1.2
     debug = 7
+    sslVersion = TLSv1.2
```
Comment (alaypatel07) [inline pkg/controller/directvolumemigration/stunnel.go:101]: just out of curiosity, was this the reason why previous tries of making proxy work with stunnel failing?

File: pkg/controller/directvolumemigration/stunnel.go [hunk 4]
```diff
@@ -51,12 +63,27 @@ data:
...
     syslog = no
     [rsync]
-    accept = {{ .StunnelPort}}
+    accept = {{ .StunnelPort }}
     CAFile = /etc/stunnel/certs/ca.crt
     cert = /etc/stunnel/certs/tls.crt
+{{ if not (eq .ProxyHost "") }}
+    protocol = connect
+    connect = {{ .ProxyHost }}
+    protocolHost = {{ .RsyncRoute }}:443
```
Comment (alaypatel07) [inline pkg/controller/directvolumemigration/stunnel.go:72]: missing this if missing an`{{ end }}`  or am I misreading?

Comment (pranavgaikwad) [reply pkg/controller/directvolumemigration/stunnel.go:72]: youre misreading. the {{ end }} is after the else block

File: pkg/controller/directvolumemigration/stunnel.go (hunk 3 above)
Comment (pranavgaikwad) [reply pkg/controller/directvolumemigration/stunnel.go:101]: @alaypatel07 nope. this was not. This actually had nothing to do with that wrong version error

**Review by alaypatel07 — APPROVED** [review]
VISIACK

**Pull request merged**